libnavigation-util/src/main/java/com/mapbox/navigation/utils/timer/MapboxTimer.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.utils.timer
+
+import com.mapbox.navigation.utils.thread.ThreadController
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * Schedules a delay of [restartAfter] seconds and then restarts.
+ *
+ * @param restartAfter Time delay until the timer should restart
+ * @param listener Hook to receive the events from [MapboxTimerListener]
+ */
+class MapboxTimer(private val restartAfter: Long, private val listener: MapboxTimerListener) {
+    private val mainControllerJobScope = ThreadController.getMainScopeAndRootJob()
+
```

Comment:
start() should acquire the job from the ThreadController, because you are canceling it in stop(). Otherwise, you'll have a dead job and when you call start() nothing will happen. In short:
private val mainControllerJobScope:JobControl? = null
start() {
  mainControllerJobScope = ThreadController.getMainScopeAndRootJob()
  <your code here>
}
another way of doing it , is instead of calling cancel() on the job do this:
mainControllerJobScope.job.children.forEach{ job ->
   job.cancel()
}
That way you can keep mainControllerJobScope as non-null

Reply:
👍